common/reactive/src/main/java/io/helidon/common/reactive/Multi.java

```diff
@@ -178,6 +181,43 @@
         return new MultiFromStream<>(stream);
     }
 
+    /**
+     * Create a {@link Multi} instance that publishes {@link ByteBuffer}s from the given {@link InputStream}.
+     * <p>
+     * If {@link InputStream} can be trusted not to block on read operations, use faster
+     * {@link Multi#from(java.io.InputStream, int)}.
+     * <p>
+     * The operator calls {@link InputStream#close()} when the cancel signal is intercepted.
+     *
+     * @param inputStream the Stream to publish
+     * @param buffSize size of the chunks to be sent to downstream in the form of {@link ByteBuffer}
+     * @param executor executor to use for waiting at {@link InputStream}'s blocking reads
+     * @return Multi
+     * @throws NullPointerException if {@code stream} is {@code null}
+     */
+    static Multi<ByteBuffer> from(InputStream inputStream, int buffSize, ExecutorService executor) {
+        Objects.requireNonNull(inputStream, "stream is null");
+        return new MultiFromInputStream(inputStream, buffSize, executor);
+    }
```

Comment:
@tomas-langer Added both variants so user can keep performance up in case he can trust `InputStream` to not block(for example `ByteArrayInputStream`) or utilize his own `ExecutorServic`e in case blocking of read operations is expected.

Reply:
Our internal rules do not allow `null` parameters - please check the executor is not null in this case

Reply:
fixed